### ISSUE TITLE
TTIR broadcast doesn't handle zero dimensions in shapes properly

### ISSUE THREAD

**Issue by ppadjinTT** [tenstorrent/tt-mlir#6975]
### Describe the bug

When a pattern appears where we broadcast a shape that has a zero-dim like here `%14 = "ttir.broadcast"(%13) <{broadcast_dimensions = array<i64: 0, 1>}> : (tensor<1x3840xbf16>) -> tensor<0x3840xbf16> loc(#loc17)`, this causes division by zero in tt-mlir and therefore a segfault.

It occurs in this section of `include/ttmlir/Utils.h`:
```
template <typename T>
inline llvm::SmallVector<T>
getBroadcastDimensions(llvm::ArrayRef<int64_t> inputShape,
                       llvm::ArrayRef<int64_t> outputShape) {
  assert(inputShape.size() == outputShape.size() &&
         "Input and Output shape should match.");

  llvm::SmallVector<T> broadcastShape;
  for (size_t i = 0; i < outputShape.size(); i++) {
    T d = outputShape[i] / inputShape[i];
    broadcastShape.push_back(d);
  }

  return broadcastShape;
}
```

### Context:
This issue is part of the bringup of Z Image Turbo transformer model

### From tt-mlir

Using built version of tt-mlir (commit sha `437870b97`) inside tt-xla:

`./third_party/tt-mlir/src/tt-mlir/build/bin/ttmlir-opt --ttir-to-ttnn-backend-pipeline="optimization-level=0 mock-system-desc-arch=blackhole mesh-shape=1,1" /path/to/graph.ttir`
Where .ttir file is:

```
#loc1 = loc("p0.1")
#loc2 = loc("p1.3")
#loc3 = loc("p2.18")
#loc4 = loc("p3.19")
#loc5 = loc("p4.21")
#loc6 = loc("p5.39")
module @SyncTensorsGraph.42 attributes {mhlo.cross_program_prefetches = [], mhlo.input_output_alias = [], mhlo.is_dynamic = false, mhlo.use_auto_spmd_partitioning = false, ttcore.meshes = #ttcore.meshes<[<"mesh" = 1x1>]>} {
  ttcore.device_module {
    builtin.module @SyncTensorsGraph.42 attributes {mhlo.cross_program_prefetches = [], mhlo.input_output_alias = [], mhlo.is_dynamic = false, mhlo.use_auto_spmd_partitioning = false, ttcore.meshes = #ttcore.meshes<[<"mesh" = 1x1>]>} {
      func.func @main(%arg0: tensor<1x3840xbf16> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<1x3840xbf16>>} loc("p0.1"), %arg1: tensor<0x1xi64> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<0x1xi64>>} loc("p1.3"), %arg2: tensor<3840xbf16> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<3840xbf16>>} loc("p2.18"), %arg3: tensor<3840x64xbf16> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<3840x64xbf16>>} loc("p3.19"), %arg4: tensor<4096x64xbf16> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<4096x64xbf16>>} loc("p4.21"), %arg5: tensor<4096x3xi32> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<4096x3xi32>>} loc("p5.39")) -> (tensor<4096x3840xbf16> {ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<4096x3840xbf16>>}, tensor<4096x3840xbf16> {ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<4096x3840xbf16>>}, tensor<4096x3xi32> {ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<4096x3xi32>>}) {
        %0 = "ttir.constant"() <{value = dense<4096> : tensor<0xi64>}> : () -> tensor<0xi64> loc(#loc)
        %1 = "ttir.constant"() <{value = dense<0> : tensor<0xi64>}> : () -> tensor<0xi64> loc(#loc)
        %2 = "ttir.permute"(%arg3) <{permutation = array<i64: 1, 0>}> : (tensor<3840x64xbf16>) -> tensor<64x3840xbf16> loc(#loc7)
        %3 = "ttir.dot_general"(%arg4, %2) <{batch_dims_lhs = array<i64>, batch_dims_rhs = array<i64>, contract_dims_lhs = array<i64: 1>, contract_dims_rhs = array<i64: 0>}> : (tensor<4096x64xbf16>, tensor<64x3840xbf16>) -> tensor<4096x3840xbf16> loc(#loc8)
        %4 = "ttir.reshape"(%arg2) <{shape = [1 : i32, 3840 : i32]}> : (tensor<3840xbf16>) -> tensor<1x3840xbf16> loc(#loc9)
        %5 = "ttir.broadcast"(%4) <{broadcast_dimensions = array<i64: 4096, 1>}> : (tensor<1x3840xbf16>) -> tensor<4096x3840xbf16> loc(#loc9)
        %6 = "ttir.add"(%3, %5) : (tensor<4096x3840xbf16>, tensor<4096x3840xbf16>) -> tensor<4096x3840xbf16> loc(#loc10)
        %7 = "ttir.reshape"(%arg1) <{shape = [0 : i32]}> : (tensor<0x1xi64>) -> tensor<0xi64> loc(#loc11)
        %8 = "ttir.lt"(%7, %1) : (tensor<0xi64>, tensor<0xi64>) -> tensor<0xi1> loc(#loc12)
        %9 = "ttir.add"(%7, %0) : (tensor<0xi64>, tensor<0xi64>) -> tensor<0xi64> loc(#loc13)
        %10 = "ttir.where"(%8, %9, %7) : (tensor<0xi1>, tensor<0xi64>, tensor<0xi64>) -> tensor<0xi64> loc(#loc14)
        %11 = "ttir.reshape"(%10) <{shape = [0 : i32, 1 : i32]}> : (tensor<0xi64>) -> tensor<0x1xi64> loc(#loc15)
        %12 = "ttir.reshape"(%arg0) <{shape = [3840 : i32]}> : (tensor<1x3840xbf16>) -> tensor<3840xbf16> loc(#loc16)
        %13 = "ttir.reshape"(%12) <{shape = [1 : i32, 3840 : i32]}> : (tensor<3840xbf16>) -> tensor<1x3840xbf16> loc(#loc17)
        %14 = "ttir.broadcast"(%13) <{broadcast_dimensions = array<i64: 0, 1>}> : (tensor<1x3840xbf16>) -> tensor<0x3840xbf16> loc(#loc17)
        %15 = "ttir.repeat"(%11) <{repeat_dimensions = array<i64: 1, 3840>}> : (tensor<0x1xi64>) -> tensor<0x3840xi64> loc(#loc18)
        %16 = "ttir.scatter"(%6, %15, %14) <{dim = 0 : i32, scatter_reduce_type = #ttcore.reduce_type<invalid>}> : (tensor<4096x3840xbf16>, tensor<0x3840xi64>, tensor<0x3840xbf16>) -> tensor<4096x3840xbf16> loc(#loc18)
        return %16, %16, %arg5 : tensor<4096x3840xbf16>, tensor<4096x3840xbf16>, tensor<4096x3xi32> loc(#loc)
      } loc(#loc)
    } loc(#loc)
  } loc(#loc)
} loc(#loc)
#loc = loc(unknown)
#loc7 = loc("transpose.20")
#loc8 = loc("dot.23")
#loc9 = loc("broadcast.27")
#loc10 = loc("add.28")
#loc11 = loc("reshape.5")
#loc12 = loc("compare.14")
#loc13 = loc("add.11")
#loc14 = loc("select.15")
#loc15 = loc("reshape.16")
#loc16 = loc("reshape.2")
#loc17 = loc("broadcast.32")
#loc18 = loc("scatter.36")
```

### From tt-xla

This is a (non-minimal) repro 
```
import torch
from diffusers import ZImageTransformer2DModel


def load_transformer():
    transformer = ZImageTransformer2DModel.from_pretrained(
        "Tongyi-MAI/Z-Image-Turbo",
        subfolder="transformer",
        torch_dtype=torch.bfloat16,
    )
    transformer.eval()
    return transformer


def load_transformer_inputs():
    latent = torch.randn(16, 1, 128, 128, dtype=torch.bfloat16)
    timestep = torch.tensor([0.5], dtype=torch.bfloat16)
    cap_feat = torch.randn(10, 2560, dtype=torch.bfloat16)
    return {
        "x": [latent],
        "t": timestep,
        "cap_feats": [cap_feat],
    }


if __name__ == "__main__":
    import torch_xla
    import torch_xla.runtime as xr
    xr.set_device_type("TT")

    transformer = load_transformer()
    inputs = load_transformer_inputs()

    # eager
    with torch.no_grad():
        output = transformer(inputs["x"], inputs["t"], inputs["cap_feats"])

    transformer.compile(backend="tt")
    transformer = transformer.to(torch_xla.device())
    inputs["x"][0] = inputs["x"][0].to(torch_xla.device())
    inputs["t"] = inputs["t"].to(torch_xla.device())
    inputs["cap_feats"][0] = inputs["cap_feats"][0].to(torch_xla.device())
    with torch.no_grad():
        output = transformer(inputs["x"], inputs["t"], inputs["cap_feats"])
    torch_xla.sync()
```

### Expected behavior

broadcasting a zero dim should be a no-op
